errors.go

```diff
@@ -50,6 +50,9 @@ type ValidationError struct {
 	// that failed to satisfy
 	SchemaPtr string
 
+	// Context represents error context for this specific validation error.
+	Context validationErrorContext
```

Comment:
field is exported, but not its type

Reply:
SchemaPtr and InstancePtr are already providing error context. Adding another field named Context is misleading.
may be `Details` is better name

Reply:
The only other logical here is `interface{}`, but using the private type helps avoid some internal type assertions. Usage of this field would be, for example:

```go
			if err.Context == nil {
				// no context
			}
			switch ctx := err.Context.(type) {
			case *jsonschema.ValidationRequiredError:
				for _, required := range ctx.MissingPropertyPtrs {
					// do whatever
				}
			}
```

Reply:
I agree, `Context` is not very useful. Details would be indeed better.